api/PayrollProcessor.Functions.Api/Features/Resources/ResourcesTrigger.cs

```diff
@@ -0,0 +1,101 @@
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+using Ardalis.GuardClauses;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Azure.Cosmos;
+using Microsoft.Azure.WebJobs;
+using Microsoft.Azure.WebJobs.Extensions.Http;
+using Microsoft.Extensions.Logging;
+using PayrollProcessor.Data.Persistence.Features.Departments;
+using PayrollProcessor.Data.Persistence.Features.Employees;
+using PayrollProcessor.Infrastructure.Seeding.Features.Employees;
+using PayrollProcessor.Infrastructure.Seeding.Features.Generators;
+
+using static PayrollProcessor.Data.Persistence.Infrastructure.Clients.AppResources.CosmosDb;
+
+namespace PayrollProcessor.Functions.Api.Features.Resources
+{
+    public class ResourcesTrigger
+    {
+        private readonly CosmosClient client;
+
+        public ResourcesTrigger(CosmosClient client)
+        {
+            Guard.Against.Null(client, nameof(client));
+
+            this.client = client;
+        }
+
+        [FunctionName(nameof(CreateResources))]
+        public async Task<ActionResult> CreateResources(
+            [HttpTrigger(AuthorizationLevel.Anonymous, "POST", Route = "resources")] HttpRequest req,
+            ILogger log)
+        {
+            log.LogInformation($"Creating all tables and queues: [{req}]");
+
+            var dbResponse = await client.CreateDatabaseIfNotExistsAsync(Databases.PayrollProcessor.Name);
+
+            await dbResponse.Database.CreateContainerIfNotExistsAsync(
+                new ContainerProperties(Databases.PayrollProcessor.Containers.Employees, partitionKeyPath: "/partitionKey"));
```

Comment:
Every collection in Cosmos needs a partition key - this is a property on the JSON documents that Cosmos uses to group data together (for horizontal scale).

Cosmos is able to query on items in the same partition of a collection (identified by the partition key property value) compared to querying across multiple partitions in the same collection.

I saw a recommendation to partition every collection on `"partitionKey"` and then add a property to every Document inserted into Cosmos called `partitionKey` instead of each collection having its own unique partition property (like `customerId`, `departmentName`, `creationDate`).